### ISSUE TITLE
TestExceptionInfo.test_traceback_not_referenced fails with pypy3

### ISSUE THREAD

**Issue by mgorny** [jaraco/keyring#442]
**Describe the bug**
`TestExceptionInfo.test_traceback_not_referenced` fails when the test suite is run using PyPy3:

```python
_________________________________________________ TestExceptionInfo.test_traceback_not_referenced _________________________________________________

self = <test_errors.TestExceptionInfo object at 0x00007fb05c3fbb40>

    def test_traceback_not_referenced(self):
        """
        Ensure that an ExceptionInfo does not keep a reference
        to the traceback, as doing so can create unintended
        side effects. See #386 for more info.
        """
        state = mock.MagicMock()
        _ = errors.ExceptionInfo(None, None, DeletionDetector(state))
>       assert state.deleted
E       AssertionError: assert False
E        +  where False = <MagicMock id='140395438949168'>.deleted

tests/test_errors.py:24: AssertionError
```

**To Reproduce**
1. Install dependencies for pypy3 (`tox` doesn't seem to work here, I used distribution packages)
2. `pypy3 -m pytest`

**Expected behavior**
All tests passing.

**Environment**

- OS: Gentoo Linux

```
$ pypy3 --version
Python 3.6.9 (?, Apr 10 2020, 20:05:19)
[PyPy 7.3.1 with GCC 9.2.0]
```

Run from git master (fb74a37d54657b279c76360cf82b850ba33d1654).

**Additional context**
My guess would be that it's due to different GC behavior in PyPy.

**Comment by jaraco** [comment]
>     `tox` doesn't seem to work here

This finding was a surprise to me. So it seems that neither black no mypy install on pypy (due to python/typed_ast#111). I've [reached out](https://gitter.im/python/typing?at=5f4143d4c712cc39e6097a12) to the typing team on gitter to see how permanent that condition is.

**Comment by jaraco** [comment]
I've merged 15f6272d1eb253940f82737b4f340365ee9879a9, which addresses the tox issue.
